src/components/Toast/Toast.tsx

```diff
@@ -1,93 +1,71 @@
    */
-  iconTitle?: string;
+  variant: Variant;
   /**
-   * Stylistic variations for the toast type.
-   * - **success** - results in a green toast
-   * - **warning** - results in a yellow toast
-   * - **error** - results in a red toast
-   * - **info** - results in a blue toast
+   * Callback when Toast is dismissed.
    */
-  variant?: 'success' | 'warning' | 'error' | 'info' | 'brand';
-}
+  onDismiss?: () => void;
+};
 
 /**
- * Primary UI component for user interaction
+ * ```ts
+ * import {Toast} from "@chanzuckerberg/eds-components";
+ * ```
+ *
+ * A toast used to provide information on the state of the page, usually in response to a
+ * user action. Ex: The user updates their profile, and a toast pop-up informs them that the
+ * data was successfully saved.
  */
 export const Toast = ({
-  iconTitle,
-  className,
   children,
-  iconName,
-  dismissible = true,
+  className,
   variant,
-  closeButtonText = 'Close',
+  onDismiss,
+  // Allow for additional attributes such as aria roles
   ...other
 }: Props) => {
   const componentClassName = clsx(
-    styles['toast'],
     className,
+    styles['toast'],
     variant === 'success' && styles['toast--success'],
-    variant === 'warning' && styles['toast--warning'],
-    variant === 'error' && styles['toast--error'],
-    variant === 'brand' && styles['toast--brand'],
+    variant === 'alert' && styles['toast--alert'],
   );
-  const [dismissed, setDismissed] = useState(false);
-
-  function onDismiss(e: any) {
-    e.preventDefault();
-    setDismissed(true);
-  }
-  return dismissed ? null : (
-    <div className={componentClassName} role="alert" {...other}>
-      {iconName && (
-        <Icon
-          name={iconName}
-          title={iconTitle}
-          className={styles['toast__icon']}
-        />
-      )}
-      <div className={styles['toast__body']}>{children}</div>
-
-      {dismissible && (
-        <Button
-          className={styles['toast__close-btn']}
-          variant="bare"
-          aria-label={closeButtonText}
-          iconName="close"
-          iconPosition="after"
-          onClick={(e: any) => onDismiss(e)}
-        />
-      )}
+  return (
+    <div className={componentClassName} {...other}>
+      <div className={styles['toast__content']}>
+        {/* TODO: point to internal NotificationIcon once fully migrated */}
+        <svg fill="currentColor" width="16px" height="16px">
+          <path d="M20 6h-1v8c0 .55-.45 1-1 1H6v1c0 1.1.9 2 2 2h10l4 4V8c0-1.1-.9-2-2-2zm-3 5V4c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v13l4-4h9c1.1 0 2-.9 2-2z" />
+        </svg>
+        {/* <NotificationIcon variant={variant} /> */}
+        <p className={styles['toast__text']}>{children}</p>
+      </div>
+      {/* TODO: point to internal CloseButton once fully migrated */}
+      {/* {onDismiss && <CloseButton color={variant} onClose={onDismiss} />} */}
+      {onDismiss && <Button onClick={onDismiss}>close</Button>}
```

Comment:
Could you use the close or x icon or whatever it is right now instead of text?

Reply:
Yup, will do